### ISSUE TITLE
Unclear why Credentials Provider fails with Database Adapter

### ISSUE THREAD

**Issue by swordensen** [nextauthjs/next-auth#10966]
### What is the improvement or update you wish to see?

The documentation found here https://authjs.dev/getting-started/authentication/credentials and here https://authjs.dev/getting-started/providers/credentials

Does not include any warnings about trying to use credentials provider with a database adapter. 

The fact that it's not supported is totally okay, especially considering the stance against using the Credentials provider.

That being said, email/password login is required for my app and it took me several hours to understand why the session object was not being retrieved. 

Also next-auth fails setting the session token silently. The only way I discovered the source of my problems is when I turned on debug mode and found this error message:

`UnsupportedStrategy: Signing in with credentials only supported if JWT strategy is enabled. Read more at https://errors.authjs.dev#unsupportedstrategy`

### Is there any context that might help us understand?

Basically, for some reason credentials provider only works with JWT tokens and not database session tokens. 

Interestingly, this information is available here: https://next-auth.js.org/providers/credentials but, not on the authjs website. I honestly did not know they were connected. 

### Does the docs page already exist? Please link to it.

https://authjs.dev/getting-started/authentication/credentials

**Comment by patelvivekdev** [comment]
Is their anyway around here?

Like I want to combine OAuth with Credentials, I manage to do that but it only works with JWT.

**Comment by julianortlieb** [comment]
The ai of the documentation has following awnser:

![image](https://github.com/nextauthjs/next-auth/assets/61797558/5a0a8bb1-2138-40f1-86f9-0d4d481703a5)

But I don't understand this argument, because you can decrypt or hash passwords?

**Comment by swordensen** [comment]
To be clear; I did not make this issue to discuss the merits of the decision made by the auth.js maintainers. I trust they have good reasons for not supporting credentials and database stored sessions. 

I just think it should be _very_ clearly documented because it is easy to assume that it would be supported otherwise.

**Comment by ThangHuuVu** [comment]
thanks for the feedback, actually it's a bit deeper in the documentation, you have to look at this page: https://authjs.dev/reference/core/providers/credentials#default-1

Hope that the explanation is clear enough! 

>It comes with the constraint that users authenticated in this manner are not persisted in the database, and consequently that the Credentials provider can only be used if JSON Web Tokens are enabled for sessions.
The functionality provided for credentials-based authentication is intentionally limited to discourage the use of passwords due to the inherent security risks of the username-password model
...

**Comment by swordensen** [comment]
@ThangHuuVu I understand that the notice is available deeper in the documentation.

Maybe I was not clear. I made this issue because I think the warning should be included in the credentials auth guide. Specifically the pages I linked above.


I think most new auth.js users will be following the guide and because the warning is not mentioned and that using database sessions with credentials fails silently by default people will continue to be confused like myself.

**Comment by ThangHuuVu** [comment]
@swordensen that's fair, PR is welcome to improve the Getting Started doc 🙌 for the “silently” part, I’d say we already have the [debugging guide](https://authjs.dev/guides/debugging) at the top of our Guide submenu, so that should be enough 🤔

**Comment by swordensen** [comment]
> @swordensen that's fair, PR is welcome to improve the Getting Started doc 🙌 for the “silently” part, I’d say we already have the [debugging guide](https://authjs.dev/guides/debugging) at the top of our Guide submenu, so that should be enough 🤔

Okay! I am more than happy to make a PR :)

I will reference this issue though. I'm not sure if it should remain closed

**Comment by swordensen** [comment]
done and done. I agree that the failing silently is documented well enough! I just added a small comment to the credentials section. 

I think these small changes would be sufficient to prevent others from going down the wrong path like I did.

**Comment by Abhishek21k** [comment]
https://www.youtube.com/watch?v=rZ-WNsxu17s

check this video

**Comment by lovasoa** [comment]
Wow, I spent way too long trying to debug this before finding this issue. I think there should be a big warning in https://authjs.dev/getting-started/providers/credentials

**Comment by mehrizi** [comment]
Hello,
As a newcommer to nextauth this limitation really sucks because for business needs we can not leverage gurus in the auth business, but we need a simple OTP (sms code and verify it) using Credential providers. For many practical reasons I decided to use database as provider and strategy. Now the custom OTP strategy fails with the original message here!

This is a simple auth scenario for a lot of startups. They even dont want to taggle with mail server as SMS providers are easier to work with.

I am trying my best not switch to the branch I handle everything at code level with simple IronSession.

Is there an abstract provider like credentials that I can achieve this with using database as strategy?

**Comment by Abhishek21k** [comment]
> Hello,
> As a newcommer to nextauth this limitation really sucks because for business needs we can not leverage gurus in the auth business, but we need a simple OTP (sms code and verify it) using Credential providers. For many practical reasons I decided to use database as provider and strategy. Now the custom OTP strategy fails with the original message here!
> 
> This is a simple auth scenario for a lot of startups. They even dont want to taggle with mail server as SMS providers are easier to work with.
> 
> I am trying my best not switch to the branch I handle everything at code level with simple IronSession.
> 
> Is there an abstract provider like credentials that I can achieve this with using database as strategy?

Instead of database strategy use the jwt one and then implement your own database session logic in the callbacks and authorize functions. This way i can do any type of auth using anything
